src/lexer/lexer.cc

```diff
@@ -0,0 +1,35 @@
+#include "lexer/lexer.h"
+
+#include <fstream>
+#include <sstream>
+
+namespace lexer {
+
+namespace {
+
+std::istream* get_stream_from_source(const std::string& source, Lexer::SourceTag tag) {
```

Comment:
Please return a unique_ptr or indicate clearly that the function gives the ownership to the caller.